kotlin/src/main/java/app/rive/runtime/kotlin/RiveAnimationView.kt

```diff
@@ -295,17 +296,17 @@ class RiveAnimationView(context: Context, attrs: AttributeSet? = null) : View(co
      */
     fun setRiveResource(
         @RawRes resId: Int,
-        artboardName: String? = drawable.artboardName,
-        animationName: String? = drawable.animationName,
+        artboardName: String? = null,
+        animationName: String? = null,
         autoplay: Boolean = drawable.autoplay,
-        fit: Fit = drawable.fit,
-        alignment: Alignment = drawable.alignment,
-        loop: Loop = drawable.loop,
+        fit: Fit = Fit.CONTAIN,
+        alignment: Alignment = Alignment.CENTER,
+        loop: Loop = Loop.NONE,
     ) {
         resourceId = resId
-        val file = File(resources.openRawResource(resId).readBytes())
-        setRiveFile(
-            file,
+        val bytes = resources.openRawResource(resId).readBytes()
+        setRiveBytes(
+            bytes,
             fit = fit,
```

Comment:
So I've moved resetting the drawable's defaults to this level; see comments on the ```setRiveFile``` function. As we eat the defaults the first time we call setRiveXXX, I just copied the defaults out to here and ```setRiveBytes```. We could have the defaults as class immutables on RiveDrawable if we wanted to keep things nicely encapsulated.